### PR TITLE
Renovate: Disable non-breaking updates for rust

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [X] Other
```

## Objective

Configure renovate to not propose non-breaking changes of dependencies. To avoid always updating to the latest versions since Bitwarden will act as a library. (We will get the latest versions through the lockfile refresh)

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)

### CODE REVIEW

File: .github/renovate.json [hunk 1]
```diff
@@ -1,19 +1,34 @@
 {
   "$schema": "https://docs.renovatebot.com/renovate-schema.json",
-  "extends": ["config:base", "schedule:monthly", ":maintainLockFilesWeekly"],
+  "extends": [
+    "config:base",
+    "schedule:monthly",
+    ":dependencyDashboard",
+    ":maintainLockFilesWeekly"
```
[the remark this answers is not in the document]
Comment (MGibson1) [reply .github/renovate.json:7]: >":pinAllExceptPeerDependencies",

We don't want to pin until we're ready. This isn't an end-user application (except for CLI) like the other repos. Pinning can break our interface with any application that wants to use these crates.

If we want to enable pinning, we need to do the work to identify public vs private dependencies and mark them as such in our various cargo.toml files

[the remark this answers is not in the document]
Comment (withinfocus) [reply .github/renovate.json:7]: Yes that's right, we talked about not pinning in this repo's case.

[the remark this answers is not in the document]
Comment (Hinton) [reply .github/renovate.json:7]: `prConcurrentLimit10` once we have more mac runners. It still takes 5h+ to run any job in our repos.

File: .github/renovate.json [hunk 2]
```diff
@@ -1,19 +1,34 @@
...
+  ],
   "separateMajorMinor": true,
+  "enabledManagers": ["cargo", "github-actions", "npm", "nuget"],
   "packageRules": [
     {
+      "groupName": "npm minor",
       "matchManagers": ["npm"],
       "matchPackagePatterns": ["*"],
```
[the remark this answers is not in the document]
Comment (Hinton) [reply .github/renovate.json:15]: It's from the template.

[the remark this answers is not in the document]
Comment (withinfocus) [reply .github/renovate.json:19]: I don't know what the UI did here, but I meant `matchPackagePatterns`.

**Review by withinfocus — APPROVED** [review]
This looks right! Let's see how it performs.

**Pull request merged**